rio_tiler/io/rasterio.py

```diff
@@ -501,6 +501,8 @@ def feature(
         height: Optional[int] = None,
         width: Optional[int] = None,
         buffer: Optional[NumType] = None,
+        all_touched: Optional[bool] = False,
+        apply_cutline: Optional[bool] = True,
```

Comment:
🤔 This adds a bit of complexity because now you have to know if your data is of multiple resolution or not 🤷 

IMO for `feature` we should always have the mask representing the combination of the internal nodata/alpha and the given cutline geometry. 

I think we could handle this specific case within the `ImageData.create_from_list` method by rescaling the cutline mask and merging all mask in a 1 band layer that should save the maximum extent of all the masks

Reply:
You are absolutely right! Just pushed a solution that works and even reduced the number of lines overall 🤓

Reply:
Just saw that after running black on it the line count increased ➕ , but its still a way simpler solution I think